### PR TITLE
MILAB-6530: more informative postfix for linked columns headers

### PR DESCRIPTION
**Pull request by AStaroverov**
<h3>Greptile Summary</h3>

This PR replaces the old single-pass linker-suffix approach in `deriveDistinctLabels` with a two-phase architecture: phase 1 (`deriveStems`) produces a minimal "stem" per column from its label, trace, and hit/anchor qualifications (no linker info), and phase 2 (`derivePostfixes`, new `linked_column_postfix.ts`) appends a "via …" postfix only where stems collide, choosing the minimal difference of source axes and linker chains. It also adds `repairBareByPresence` to fix columns that were previously distinguished only by absence of a peer's token, and narrows `discoverLabelColumns` to primary columns with `maxHops: 0`.

- **New `linked_column_postfix.ts`**: Implements phase 2 — groups entries by stem, escalates over root axis → linker chain slots until unique, exposes a richer `LinkerFormatter` API (`{ root, linkers }` instead of `string[]`).
- **`derive_distinct_labels.ts` refactored**: `LINKER_TYPE` synthetic trace entry and `dropRedundantLinkerSuffix` removed; new `repairBareByPresence` added; `LinkerStep.spec` narrowed from `PObjectSpec` to `PColumnSpec`.
- **`discoverLabelColumns` narrowed**: Only primary columns are used as anchors, `maxHops` hardcoded to `0` — prevents linked/secondary columns from being re-discovered as label columns.

<details open><summary><h3>Confidence Score: 4/5</h3></summary>

Safe to merge after removing the dead debug data blob from createPlDataTableV3.ts — everything else is well-tested and logically sound.

The two-phase label derivation logic and the new postfix module are thoroughly tested and the algorithm is well-reasoned. One file, createPlDataTableV3.ts, contains a 107-line constant assigned to `_` that holds real-looking block IDs and serialized column specs — dead code that would ship in the built package as-is.

sdk/model/src/components/PlDataTable/createPlDataTable/createPlDataTableV3.ts — the const _ = [...] block at the bottom must be deleted before merging.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| sdk/model/src/labels/linked_column_postfix.ts | New phase-2 postfix module: derives minimal "via …" suffixes for colliding linked columns by comparing root axes and linker chains; well-documented with a complete test suite covering all edge cases. |
| sdk/model/src/labels/derive_distinct_labels.ts | Refactored into two phases: deriveStems (phase 1, label+trace minimization, formerly the whole function) and derivePostfixes (phase 2, delegated to linked_column_postfix); removed LINKER_TYPE synthetic trace entry; added repairBareByPresence to un-bare columns distinguished only by absence. |
| sdk/model/src/components/PlDataTable/createPlDataTable/createPlDataTableV3.ts | discoverLabelColumns call narrowed to primary-only columns; contains a 107-line dead debug data blob (const _ = [...]) that must be removed before merging. |
| sdk/model/src/components/PlDataTable/createPlDataTable/discoverColumns.ts | discoverLabelColumns simplified: now uses only primary columns as anchors with maxHops=0 (direct-only), removing the secondary-column pass and computed maxHops; return type of discoverTableColumns widened to always-defined. |
| sdk/model/src/labels/derive_distinct_labels.test.ts | Tests updated for the two-phase architecture: new regression tests for "cluster-id preset" and "by-presence repair", updated expectations for minimal linker rendering, two qualification tests deferred to .todo. |
| sdk/model/src/labels/linked_column_postfix.test.ts | New test file covering derivePostfixes: root-only, linker-only, mixed-group, domain-qualified, no-collision, and collision-subset cases; one .todo deferred for per-row trimming. |
| etc/blocks/table-test/model/src/index.ts | Custom linker formatter updated from (string[]) to ({root, linkers}) shape to match the new LinkerFormatter API. |
| sdk/model/src/columns/column.ts | Cosmetic: import style changed from import type { ... } to import { type ... } per-identifier; no functional change. |

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
sequenceDiagram
    participant Caller
    participant DDL as deriveDistinctLabels
    participant DS as deriveStems (phase 1)
    participant DP as derivePostfixes (phase 2)
    participant RG as resolveGroup
    participant RPB as repairBareByPresence

    Caller->>DDL: entries[], options
    DDL->>DS: entries[], options
    DS->>DS: enrichRecord() — build trace (label + qual types, no linker)
    DS->>DS: minimizeTypeSet() — minimal discriminating type set
    DS->>DS: build() — render labels from minimized set
    DS->>RPB: rendered[], minimized, records, stats
    RPB->>RPB: find bare columns (distinguished only by absence)
    RPB->>RPB: bestDistinguishingType() per bare column
    RPB-->>DS: patched stems[]
    DS-->>DDL: stems[]
    DDL->>DP: "PostfixEntry[] {stem, hit, linkers[]}"
    DP->>DP: group entries by stem
    loop per colliding stem-group
        DP->>RG: group entries + indices
        RG->>RG: extractRoot() — source axis of linkers[0]
        RG->>RG: escalate slots (root → linker[0] → linker[1]…) until allUnique
        RG->>RG: drop redundant slots (greedy)
        RG->>RG: "renderAll() — format({ root, linkers }, hit, idx)"
        RG-->>DP: postfix strings[]
    end
    DP-->>DDL: final labels[] (stem + postfix or bare stem)
    DDL-->>Caller: string[]
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
sequenceDiagram
    participant Caller
    participant DDL as deriveDistinctLabels
    participant DS as deriveStems (phase 1)
    participant DP as derivePostfixes (phase 2)
    participant RG as resolveGroup
    participant RPB as repairBareByPresence

    Caller->>DDL: entries[], options
    DDL->>DS: entries[], options
    DS->>DS: enrichRecord() — build trace (label + qual types, no linker)
    DS->>DS: minimizeTypeSet() — minimal discriminating type set
    DS->>DS: build() — render labels from minimized set
    DS->>RPB: rendered[], minimized, records, stats
    RPB->>RPB: find bare columns (distinguished only by absence)
    RPB->>RPB: bestDistinguishingType() per bare column
    RPB-->>DS: patched stems[]
    DS-->>DDL: stems[]
    DDL->>DP: "PostfixEntry[] {stem, hit, linkers[]}"
    DP->>DP: group entries by stem
    loop per colliding stem-group
        DP->>RG: group entries + indices
        RG->>RG: extractRoot() — source axis of linkers[0]
        RG->>RG: escalate slots (root → linker[0] → linker[1]…) until allUnique
        RG->>RG: drop redundant slots (greedy)
        RG->>RG: "renderAll() — format({ root, linkers }, hit, idx)"
        RG-->>DP: postfix strings[]
    end
    DP-->>DDL: final labels[] (stem + postfix or bare stem)
    DDL-->>Caller: string[]
```

</a>
</details>

<sub>Reviews (2): Last reviewed commit: ["Refactor label derivation logic to enhan..."](https://github.com/milaboratory/platforma/commit/f25c08f31503d981bc1d75b86b7cf24c74ae198b) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=45109812)</sub>

### CODE REVIEW

File: sdk/model/src/labels/linked_column_postfix.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+function defaultLinkerFormatter(parts: LinkerParts): string {
+  const chain = parts.linkers.map((l) => l.text).join(" > ");
+  const pieces = [parts.root?.text, chain || undefined].filter((p): p is string => !isNil(p));
+  return pieces.length === 0 ? "" : `via ${pieces.join(" ")}`;
+}
+
+// --- Node identity & minimal token (level 2: WHAT differs) -------------------
+
+type Identity = {
```
Comment (vadimpiven) [inline sdk/model/src/labels/linked_column_postfix.ts]: why no type?

**Review by vadimpiven — APPROVED** [review]

**Comment by AStaroverov** [comment]
@greptileai
